### DISCUSSION TITLE
Support for multiple libraries

### DISCUSSION THREAD

**Discussion by arifer612** [Radiokot/photoprism-android-client#129] [Ideas]
For a group with multiple users that wish to have the ability to maintain a library with private photos, it is recommended by PhotoPrism to use make use of [multiple libraries](https://docs.photoprism.app/user-guide/users/libraries/), i.e., a private library for each user and a shared library for the group. This is the approach I have taken for my family, with each member of the family using this android client to connect to our private library and the progressive web app to connect to the shared library.

What my family and I would love to do is to move away from the progressive web app and to be able to view both our own private library and the shared family library on this android client. One way I can think of this being implemented is by having "profiles" that we can switch between in the client, where each "profile" can have a different library root URL and/or user. Switching between profiles will thus allow us to switch between the private library and the shared library.

**Comment by Radiokot** [comment]
Hi. It is amazing that you have convinced your family to use 2 PhotoPrism instances 😄However, I don't want to add such complexity to the app, which also doesn't align well with other features I am planning to add. Instead, I suggest using a clone package for your other library: https://github.com/Radiokot/photoprism-android-client/releases/download/1.24.1(38)/ua.com.radiokot.photoprism.clone-1.24.1-clone.apk

**Comment by arifer612** [comment]
Thanks for getting back. Took me a while to convince them, but it's working better than I ever expected so I guess I got lucky (?) :laughing:. Sure thing, I understand that this added layer of complexity does not align with what you have in mind, but if it ever comes to a point where you might need to host multiple libraries too, I hope you'll consider this idea as a solution around it as well!

Meanwhile, I suppose I'll go with your recommendation and use a clone package for the other library. Thank you!

**Comment by tilusnet** [comment]
The lack of connection profiles is one of the main letdowns of this app. 

Having several PhotoPrism instances is indeed a fairly common pattern for several user groups and privacy considerations. 
I often find myself needing to upload pictures to several running PhotoPrism instances depending on the target user group audience. Whenever I use this app I find it extremely annoying that I am forced to edit a "global" URL+creds associated every time I upload for a different audience. It's a PITA.

**Reply by Radiokot** [reply]
Hi,
I didn't change my opinion on the library change feature. I'd like to keep the "1 app – 1 library" model. If you need to access more than two PhotoPrism libraries simultaneously, I suggest you fork the repo, enable GitHub Actions, and configure the build of multiple clones, with the example of the `releaseClone` build type in `app/build.gradle`. With that, you'll be able to get your own number of separate APKs, even with a title and icon of your desire. It's quite straightforward.

**Reply by tilusnet** [reply]
Thanks for your reply.

All the app needs to do is to remember some configuration profiles - there's no need to keep several configs active simultaneously. Only one.
